plugins/out_stackdriver/stackdriver.c

```diff
@@ -2652,10 +2656,12 @@ static int parse_partial_success_response(struct flb_http_client* c,
                                                    "@type", 5,
                                                    MSGPACK_OBJECT_STR,
                                                    &at_type);
+        strncpy(at_type_str, at_type.via.str.ptr,
+                PARTIAL_SUCCESS_GRPC_TYPE_SIZE);
         if (ret != 0 ||
-            (at_type.via.str.size == PARTIAL_SUCCESS_GRPC_TYPE_SIZE
-                && strncmp(at_type.via.str.ptr, PARTIAL_SUCCESS_GRPC_TYPE,
-                           PARTIAL_SUCCESS_GRPC_TYPE_SIZE) == 0)) {
+            at_type.via.str.size != PARTIAL_SUCCESS_GRPC_TYPE_SIZE ||
+            strncmp(at_type_str, PARTIAL_SUCCESS_GRPC_TYPE,
+                           PARTIAL_SUCCESS_GRPC_TYPE_SIZE) != 0) {
```

Comment:
Careful! This would also match responses with, e.g., `@type`==`"type.googleapis.com/google.logging.v2.WriteLogEntriesPartialErrors1234"`, because you cut of everything past 66 characters. Not that it's likely to happen in practice, but it might make sense to define `PARTIAL_SUCCESS_GRPC_TYPE_SIZE` as 67, to accommodate for the trailing `'\0'`…

Reply:
`at_type.via.str.size != PARTIAL_SUCCESS_GRPC_TYPE_SIZE` protects us from this concern.